app_feup/lib/controller/networking/print_login.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:http/http.dart' as http;
+
+//Returns the cookie for print.up.pt
+Future<String> printLogin(String studentNumber, String password) async {
+  final url1 = 'https://print.up.pt/user';
+
+  final getResponse = await http.get(Uri.parse(url1));
+
+  final url = 'https://print.up.pt/app';
+
+  final Map<String, String> headers = Map<String, String>();
+  headers['content-type'] = 'application/x-www-form-urlencoded';
+  headers['cookie'] = getResponse.headers['set-cookie'];
+  headers['origin'] = 'https://print.up.pt';
+
+  String payload =
+      'service=direct%2F1%2FHome%2F%24Form&sp=S0'
+      '&Form0=%24Hidden%240%2C%24Hidden%241%2CinputUsername%2CinputPassword'
+      '%2C%24Submit%240%2C%24PropertySelection&%24Hidden%240=true&%24'
+      'Hidden%241=X&%24Submit%240=Entrar&%24PropertySelection=pt_PT';
+  final username = 'up${studentNumber}@up.pt';
+  payload += '&inputUsername=${username}&inputPassword=${password}';
+
+  final postResponse =
+      await http.post(Uri.parse(url), headers: headers, body: payload);
```

Comment:
The body the [`post` method](https://pub.dev/documentation/http/latest/http/Client/post.html) accepts is a polymorphic one. Perhaps it would be enough to pass the username and the password fields in a map. Can you try that?

Reply:
Also, can you specify a [timeout](https://api.dart.dev/dev/2.0.0-dev.69.2/dart-async/Future/timeout.html) for this for the function to return error to the caller?